cmd/fleet/serve_test.go

```diff
@@ -23,6 +23,23 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// safeStore is a wrapper around mock.Store to allow for concurrent calling to
+// AppConfig, in the past we have seen this test fail with a data race warning.
+//
+// TODO: if we see other tests failing for similar reasons, we should build a
+// more robust pattern instead of doing this everywhere
```

Comment:
:+1: yeah there was talk in the past of doing this in our code-generation of datastore mock. For now, this is probably good enough, but as you mention, if we start needing this in multiple places, we should adjust the mock code generation directly.

Reply:
Yeah hopefully it wouldn't be too big a change to fork and slightly adjust the code generated to add the mutex. Alternatives would likely require big changes in our tests.

Reply:
sounds good to me! I'm super curious if adding a lock to every mock method will have any significant impact on how long the test suite takes to run.

I created https://github.com/fleetdm/fleet/issues/7955 to make sure this doesn't get lost.